Feat port 74.5.33

pabloarodas opened the pull request:
fix: Improve efficiency of SimpleSearchQueryConverter's parameter prefix generation
- Move AlphanumericRandomValueStringGenerator from test to main
- Use AlphanumericRandomValueStringGenerator when SimpleSearchQueryConverter generates parameter prefix instead of regenerating the string until it does not contain "-" using RandomValueStringGenerator

[#179824012]

hsinn0 committed:
fix: Improve efficiency of SimpleSearchQueryConverter's parameter prefix generation
- Move AlphanumericRandomValueStringGenerator from test to main
- Use AlphanumericRandomValueStringGenerator when SimpleSearchQueryConverter generates parameter prefix instead of regenerating the string until it does not contain "-" using RandomValueStringGenerator

[#179824012]

hsinn0 reviewed (approved):
Looks good.  Approving pending the failed CI check to pass.

ram-pivot committed:
fix: change use of RandomValueStringGenerator to
AlphanumericRandomValueStringGenerator

Co-authored-by: Ramkumar Vengadakrishnan <rvengadakrishnan@pivotal.io>
Co-authored-by: Pablo Rodas <prodas@vmware.com>

pabloarodas committed:
fix: Removed references to deprecated RandomValueStringGenerator

Replaced RandomValueStringGenerator class with
AlphanumericRandomValueStringGenerator class

Co-authored-by: Pablo Rodas <prodas@vmware.com>
Co-authored-by: Ramkumar Vengadakrishnan <ramkumarv@vmware.com>

[#182855359]

pabloarodas committed:
fix: Fixed integration tests

Removed references to deprecated RandomValueStringGenerator and replaced
it with AlphanumericRandomValueStringGenerator class

Co-authored-by: Pablo Rodas <prodas@vmware.com>
Co-authored-by: Ramkumar Vengadakrishnan <ramkumarv@vmware.com>

[#182855359]

pabloarodas committed:
feat: Fixed Sonar code smells and warnings

Co-authored-by: Pablo Rodas <prodas@vmware.com>
Co-authored-by: Ramkumar Vengadakrishnan <ramkumarv@vmware.com>

[#182855359]

hsinn0 commented on server/src/main/java/org/cloudfoundry/identity/uaa/util/AlphanumericRandomValueStringGenerator.java:
```diff
@@ -0,0 +1,44 @@
+import java.security.SecureRandom;
+import java.util.Random;
+
+public class AlphanumericRandomValueStringGenerator {
```
In 74.5.x, `AlphanumericRandomValueStringGenerator` extends `org.springframework.security.oauth2.common.util.RandomValueStringGenerator`.  Any specific reason not doing the same?

ram-pivot replied:
@hsinn0 Sonar gave warnings to not extend a class that is Deprecated and Sonar checks failed.

hsinn0 replied:
I see.  In fact, we know that the whole `spring-security-oauth2` is deprecated and have a story to replace with different one.  Because the whole UAA project has dependency to the EOL-ed `spring-security-oauth2` package, it practically does not really matter for this one class to depend on a deprecated class.
Anyhow, this change will do the job as it is, so I am fine with it then.

ram-pivot replied:
Yes agree. Since we modified this file, sonar was scanning and complained about this specific class usage.
